### DISCUSSION TITLE
Containerd or cri proxy settings

### DISCUSSION THREAD

**Discussion by eneko-perez** [siderolabs/talos#7909] [Q&A]
Hi all,

I'm trying to setup Talos behind a proxy and in the bootstrapping process I can't download any image (kubelet, etcd) so the installation can not continue and I'm in a booting permanent state. The errors I see in the logs are:

[   63.900389] [talos] retrying error: failed to pull image "ghcr.io/siderolabs/kubelet:v1.28.2": failed to resolve reference "ghcr.io/siderolabs/kubelet:v1.28.2": failed to do request: Head "https://ghcr.io/v2/siderolabs/kubelet/manifests/v1.28.2": proxyconnect tcp: dial tcp: lookup (my proxy info here) on 10.96.64.9:53: read udp 10.96.105.3:44231->10.96.64.9:53: i/o timeout

The rest of the installation has gone well so far by defining env variables in the control plane yaml. 

I tried to append this information to containerd this way:

`      - content: |
          [Service]
          Environment="HTTPS_PROXY=myproxyaddress:port"
        path: /etc/cri/conf.d/20-customization.part
        op: append`

but I'm not sure if it's working or not or I'm doing something wrong.

Besides this error, I'm getting a lot of:

`[ 3479.026763] [talos] controller failed {"component": "controller-runtime", "controller": "k8s.NodeApplyController", "error": "1 error(s) occurred:\n\terror getting node: Get \"https://localhost:6443/api/v1/nodes/talos-08x-6in?timeout=30s\": dial tcp [::1]:6443: connect: connection refused"}
[ 3482.797366] [talos] task startAllServices (1/1): service "kubelet" to be "up"
[ 3484.515127] [talos] hello failed {"component": "controller-runtime", "controller": "cluster.DiscoveryServiceController", "error": "rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing: dial tcp: lookup discovery.talos.dev on 10.96.64.9:53: read udp 10.96.105.3:38563->10.96.64.9:53: i/o timeout\"", "endpoint": "discovery.talos.dev:443"} `

Any suggestions to fix these two issues? It feels like I'm almost there but it's incredible the amount of time I'm dedicating to such a simple setup, in part due to my lack of understanding of how Talos works internally but also because of my entangled setup.

Thank you.

**Comment by steverfrancis** [comment]
Did you set https proxy in talos environment as at the bottom of
https://www.talos.dev/v1.5/reference/kernel/ ?

**Reply by eneko-perez** [reply]
I tried that too, yes.

**Reply by smira** [reply]
what is the actual error? hat you posted is unrelated to container image pull process

**Reply by eneko-perez** [reply]
The error was timing out while querying my proxy server (while pulling images). This was actually a misconfiguration issue that I resolved yesterday by setting up the right dns servers. Now I have, finally, a Talos installation working with all the components and Kubernetes too. I just need to figure out how to make it work through this proxy of hell that is acting as man in the middle but that's another story.

Thank you for your continuous support @smira
